2-0-data-structures-and-algorithms/src/test/java/com/bobocode/stack/StackTest.java

```diff
@@ -1,88 +1,355 @@
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.function.Predicate;
+
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNullPointerException;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
 @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 class StackTest {
 
+    private static final String PROPER_CLASSNAME = "Node";
+
+    private static final Predicate<Field> NODE_FIELD_PREDICATE = field -> field.getType().getSimpleName().equals(PROPER_CLASSNAME)
+            && field.getName().toLowerCase().contains("head") || field.getName().toLowerCase().contains("first");
+    private static final Predicate<Field> SIZE_FIELD_PREDICATE = field -> field.getName().toLowerCase().contains("size");
+    private static final Predicate<Field> NODE_ELEMENT_FIELD = field -> field.getName().toLowerCase().contains("element")
+            || field.getName().toLowerCase().contains("value")
+            || field.getName().toLowerCase().contains("item");
+    private static final Predicate<Field> NODE_NEXT_FIELD = field -> field.getType().getSimpleName().equals(PROPER_CLASSNAME)
+            && field.getName().toLowerCase().contains("next");
+
     private Stack<Integer> intStack = new LinkedStack<>();
 
     @Test
     @Order(1)
+    void checkProperInnerClassName() {
+        String name = getInnerClass().getSimpleName();
+        assertThat(name).isEqualTo(PROPER_CLASSNAME);
+    }
+
+    @Test
+    @Order(2)
+    void checkProperStackFieldNames() {
+        Field[] fields = LinkedStack.class.getDeclaredFields();
+
+        boolean hasSizeField = Arrays.stream(fields)
+                .anyMatch(SIZE_FIELD_PREDICATE);
+        boolean hasNodeField = Arrays.stream(fields)
+                .anyMatch(NODE_FIELD_PREDICATE);
+
+        assertThat(hasNodeField).isTrue();
+        assertThat(hasSizeField).isTrue();
+    }
+
+    @Test
+    @Order(3)
+    void checkProperNodeFields() {
+        Field[] fields = getInnerClass().getDeclaredFields();
+
+        boolean hasElement = Arrays.stream(fields)
+                .anyMatch(NODE_ELEMENT_FIELD);
+        boolean hasNext = Arrays.stream(fields)
+                .anyMatch(NODE_NEXT_FIELD);
+
+        assertThat(hasElement).isTrue();
+        assertThat(hasNext).isTrue();
+    }
+
+    @Test
+    @Order(4)
+    void ofAddElements() {
+        intStack = LinkedStack.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+
+        for (int i = 1; i <= 10; i++) {
+            assertThat(contains(i)).isTrue();
+        }
+    }
+
+    @Test
+    @Order(5)
+    void pushElements() {
+        intStack.push(55);
+        intStack.push(59);
+        intStack.push(74);
+
+        assertThat(contains(55)).isTrue();
+        assertThat(contains(59)).isTrue();
+        assertThat(contains(74)).isTrue();
+    }
+
+    @Test
+    @Order(6)
     void pushElementOntoEmptyStack() {
```

Comment:
Regarding our test name convention this test can name `pushAddsElementWhenStackIsEmpty()`